launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -401,8 +401,7 @@ boolean isClientMode(Map<String, String> userProps) {
    * Return whether the given main class represents a thrift server.
    */
   private boolean isThriftServer(String mainClass) {
-    return (mainClass != null &&
-      mainClass.equals("org.apache.spark.sql.hive.thriftserver.HiveThriftServer2"));
+    return ("org.apache.spark.sql.hive.thriftserver.HiveThriftServer2".equals(mainClass));
```

Comment:
Is this needed? It looks like we already have some defensive code for this `private boolean isThriftServer`.

https://github.com/apache/spark/blob/cb0ef06b641845be5774b18c9c24d67c6ab4561a/launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java#L235

Reply:
It's needed. If remove `mainClass != null`, pyspark test will fail.
```plain
========================================================================
Running PySpark tests
========================================================================
Running PySpark tests. Output is in /__w/spark/spark/python/unit-tests.log
Will test against the following Python executables: ['python3.9', 'pypy3']
Will test the following Python modules: ['pyspark-sql', 'pyspark-resource', 'pyspark-mllib']
python3.9 python_implementation is CPython
python3.9 version is: Python 3.9.5
pypy3 python_implementation is PyPy
pypy3 version is: Python 3.7.12 (7.3.7+dfsg-1~ppa1~ubuntu20.04, Oct 26 2021, 02:00:43)
[PyPy 7.3.7 with GCC 9.3.0]
Starting test(pypy3): pyspark.sql.tests.test_arrow (temp output: /tmp/pypy3__pyspark.sql.tests.test_arrow__2_kthvri.log)
Finished test(pypy3): pyspark.sql.tests.test_arrow (1s) ... 121 tests were skipped
Starting test(pypy3): pyspark.sql.tests.test_arrow_map (temp output: /tmp/pypy3__pyspark.sql.tests.test_arrow_map__ot8ujcj9.log)
Finished test(pypy3): pyspark.sql.tests.test_arrow_map (0s) ... 7 tests were skipped
Starting test(pypy3): pyspark.sql.tests.test_catalog (temp output: /tmp/pypy3__pyspark.sql.tests.test_catalog__67s8rrv3.log)
Exception in thread "main" java.lang.NullPointerException
	at org.apache.spark.launcher.SparkSubmitCommandBuilder.isThriftServer(SparkSubmitCommandBuilder.java:404)
	at org.apache.spark.launcher.SparkSubmitCommandBuilder.buildSparkSubmitCommand(SparkSubmitCommandBuilder.java:265)
	at org.apache.spark.launcher.SparkSubmitCommandBuilder.buildCommand(SparkSubmitCommandBuilder.java:164)
	at org.apache.spark.launcher.Main.buildCommand(Main.java:113)
	at org.apache.spark.launcher.Main.main(Main.java:63)
```